src/query/string_opfunc.c

```diff
@@ -3678,6 +3680,130 @@ db_string_trim (const MISC_OPERAND tr_operand, const DB_VALUE * trim_charset, co
+
+  /* Assert that DB_VALUE structures have been allocated. */
+  assert (string1 != (DB_VALUE *) NULL);
+  assert (string2 != (DB_VALUE *) NULL);
+  assert (result != (DB_VALUE *) NULL);
+
+  /* Categorize the two input parameters and check for errors. Verify that the parameters are both character strings.
+   * Verify that the input strings belong to compatible categories. */
+  string1_category = qstr_get_category (string1);
+  string2_category = qstr_get_category (string2);
+
+  str1_type = DB_VALUE_DOMAIN_TYPE (string1);
+  str2_type = DB_VALUE_DOMAIN_TYPE (string2);
+
+  if (!QSTR_IS_ANY_CHAR_OR_BIT (str1_type) || !QSTR_IS_ANY_CHAR_OR_BIT (str2_type))
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_QSTR_INVALID_DATA_TYPE, 0);
+      return ER_QSTR_INVALID_DATA_TYPE;
+    }
+  if (string1_category != string2_category)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_QSTR_INCOMPATIBLE_CODE_SETS, 0);
+      return ER_QSTR_INCOMPATIBLE_CODE_SETS;
+    }
+
+  /* A string which is NULL (not the same as a NULL string) is ordered less than a string which is not NULL.  Two
+   * strings which are NULL are ordered equivalently. If both strings are not NULL, then the strings themselves are
+   * compared. */
+  if (DB_IS_NULL (string1) && !DB_IS_NULL (string2))
+    {
+      cmp_result = -1;
+    }
+  else if (!DB_IS_NULL (string1) && DB_IS_NULL (string2))
+    {
+      cmp_result = 1;
+    }
+  else if (DB_IS_NULL (string1) && DB_IS_NULL (string2))
+    {
+      cmp_result = 0;
+    }
+  else if (db_get_string_codeset (string1) != db_get_string_codeset (string2))
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_QSTR_INCOMPATIBLE_CODE_SETS, 0);
+      return ER_QSTR_INCOMPATIBLE_CODE_SETS;
+    }
+  else
+    {
+      int coll_id;
+
+      switch (string1_category)
+	{
+	case QSTR_CHAR:
+	case QSTR_NATIONAL_CHAR:
+
+	  assert (db_get_string_codeset (string1) == db_get_string_codeset (string2));
+
+	  LANG_RT_COMMON_COLL (db_get_string_collation (string1), db_get_string_collation (string2), coll_id);
+
+	  if (coll_id == -1)
+	    {
+	      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_QSTR_INCOMPATIBLE_COLLATIONS, 0);
+	      return ER_QSTR_INCOMPATIBLE_COLLATIONS;
+	    }
+
+	  coll_id = db_get_string_collation (string1);
+	  assert (db_get_string_collation (string1) == db_get_string_collation (string2));
+
+	  if (!ignore_trailing_space)
+	    {
+	      ti = (QSTR_IS_FIXED_LENGTH (str1_type) || QSTR_IS_FIXED_LENGTH (str2_type));
+	    }
+	  cmp_result = QSTR_COMPARE (coll_id, DB_GET_UCHAR (string1), (int) db_get_string_size (string1),
+				     DB_GET_UCHAR (string2), (int) db_get_string_size (string2), ti);
+	  break;
+	case QSTR_BIT:
+	  cmp_result = varbit_compare (DB_GET_UCHAR (string1), (int) db_get_string_size (string1),
+				       DB_GET_UCHAR (string2), (int) db_get_string_size (string2));
+	  break;
+	default:		/* QSTR_UNKNOWN */
+	  break;
```

Comment:
"assert" may be needed.
```suggestion
	default:		/* QSTR_UNKNOWN */
	  assert (false);
	  break;
```

Reply:
OK, I fixed it.